src/providers/Web3Provider.tsx

```diff
@@ -175,8 +174,9 @@ export const Web3ContextProvider: FC<PropsWithChildren> = ({ children }) => {
         params: [{ chainId: utils.hexlify(toNetworkChainId) }],
       })
     } catch (e) {
-      // Chain is not available in the Metamask
-      if (e?.code !== 4902) {
+      // Metamask desktop - Throws e.code 4902 when chain is not available
+      // Metamask mobile - Error includes wallet_addEthereumChain & requested chainId in this case 0x5afe
+      if (e?.code !== 4902 && !(e?.message?.includes('wallet_addEthereumChain') && e?.message?.includes(utils.hexlify(toNetworkChainId)))) {
```

Comment:
I have a hard time believing 4902 was insufficient after seeing it recommended by docs https://docs.metamask.io/wallet/how-to/add-network/#example

and i found it in code too https://github.com/MetaMask/metamask-mobile/blob/f63c1ddf7637d17ac761d5b5fb897d5b0a322b4b/app/core/RPCMethods/wallet_switchEthereumChain.js#L128-L131

Reply:
It is a known issue: https://github.com/MetaMask/metamask-mobile/issues/3312. Updated the description, and switched to generic code checking.

Reply:
:scream: